R/begin.R

```diff
@@ -16,14 +17,31 @@
 begin = function(
   wd = NULL, 
   load = c( 'magrittr', 'dplyr' ),
+  keep = NULL,
   scipen = FALSE,
   verbose = TRUE,
   repos = 'http://cran.us.r-project.org'
 ){
   
   # clear workspace.
-  rm( list = ls( all.names = TRUE, envir = parent.frame() ), envir = parent.frame() )
-  if( verbose ) cat( 'Cleared variables from environment. \n' )
+  envobjs = ls( all.names = TRUE, envir = parent.frame() )
+  drop = setdiff( envobjs, keep )
+  # check if attempting to keep any objects not in environment
+  if( length( keep ) > 0 ){
+    missing = setdiff( keep, envobjs )
+    if( length( missing ) > 0 ){ 
+      stop( 
+        paste0( 'Attempting to keep objects not in environment: [', 
+        paste0( missing, collapse = ', ' ), ']. ' 
+      )) 
+    }
+  }
+  # clear vars not set to keep
+  rm( list = drop, envir = parent.frame() )
+  if( verbose ) cat( 
+    'Cleared variables from environment: [', 
```

Comment:
I'd actually rather not print all the variables since that could be a very long list that isn't helpful to the user. Can you change this back to what it was before?

Feel free to add 'except variable ...' if you'd like to make it clear to the user that their argument was accepted, since that will still avoid the very-long output.

Reply:
If you make the change and push it, the PR will automatically update.